src/hooks/usePublicMessages.js

```diff
@@ -52,102 +48,102 @@ export default function usePublicMessages(infoPoint, routes) {
-        return publicMessage.routes.some((route) => routes.includes(route.abbreviation));
-      } else {
-        return true;
-      }
-    });
+    if (!(publicMessages instanceof Array)) {
+      return publicMessages;
+    } else {
+      return normalizePublicMessages(sortPublicMessages(filterPublicMessages(publicMessages, routes)));
+    }
   }, [routes, publicMessages]);
 }
 
 /**
- * Fetches public message data from an Avail InfoPoint API.
+ * Fetches public message data from an Avail InfoPoint API. See InfoPoint Swagger UI for types.
  *
  * @param {URL} infoPoint
- * @return {Promise<[PublicMessageObject]>}
+ * @return {Promise<[{}]>}
  * @see {usePublicMessages}
  */
 async function fetchPublicMessages(infoPoint) {
-  const routesById = {};
+  const routesByID = {};
   const getAllRoutesResponse = await fetch(new URL('Routes/GetAllRoutes', infoPoint));
   const getAllRoutesJSON = await getAllRoutesResponse.json();
   getAllRoutesJSON.forEach((route) => {
-    routesById[route['RouteId']] = route;
+    routesByID[route['RouteId']] = route;
   });
 
-  const publicMessages = [];
   const getCurrentMessagesResponse = await fetch(new URL('PublicMessages/GetCurrentMessages', infoPoint));
   const getCurrentMessagesJSON = await getCurrentMessagesResponse.json();
-  getCurrentMessagesJSON.forEach((publicMessage) => {
-    let sortOrder;
-    let routes;
-    if (publicMessage['Routes'] && publicMessage['Routes'].length > 0) {
-      routes = publicMessage['Routes'].map((routeId) => {
-        if (
-          typeof(routesById[routeId]['SortOrder']) === 'number' &&
-          (typeof(sortOrder) !== 'number' || routesById[routeId]['SortOrder'] < sortOrder)
-        ) {
-          sortOrder = routesById[routeId]['SortOrder'];
-        }
-        return {
-          id: routeId,
-          abbreviation: routesById[routeId]['RouteAbbreviation'],
-          color: routesById[routeId]['Color'] || null,
-          textColor: routesById[routeId]['TextColor'] || null,
-          sortOrder: routesById[routeId]['SortOrder'] || null,
-        };
-      }).sort(compareRoutes);
+  return getCurrentMessagesJSON.map((publicMessage) => ({
+    ...publicMessage,
+    'Routes': publicMessage['Routes']?.map((routeID) => routesByID[routeID]) || [],
+  }));
+}
+
+/**
+ * Filters public messages based on a list of given route abbreviations. If the message does not apply to at least
+ * one of the given abbreviations, it will be filtered out.
+ *
+ * @param {[{}]} publicMessages - raw public message data.
+ * @param {[String]|null} routeAbbreviations
+ * @return {[{}]}
+ * @see {usePublicMessages}
+ */
+function filterPublicMessages(publicMessages, routeAbbreviations) {
+  return publicMessages.filter((message) => {
+    if ((routeAbbreviations instanceof Array) && (message['Routes'].length > 0)) {
+      return message['Routes'].some((route) => route['RouteAbbreviation'].includes(route.abbreviation));
```

Comment:
I would understand not wanting to do it here, but since you basically rewrote it all anyway.... #15 mentions removing all filtered out routes. If you want to keep that separate though then fine

<sub> separate PRs for separate issues! </sub> blahblahblah